README.md

```diff
@@ -12,176 +12,413 @@ Compartmentalization of host behavior hooks for JS.
+  // object, but will instead produce an emulation of a module exports namespace,
+  // using a frozen snapshot of the own properties of the given object.
+  // The module exports namespace for this module does not reflect any future
+  // changes to the shape of the given object.
+  | {
+    // Any object that does not pass a module exports namespace brand check:
+    namespace: ^ModuleExportsNamespace,
+  };
+
 type CompartmentConstructorOptions = {
-  // JF has a better way for:
-  //   randomHook(): number; // Use for Math.random()
-  //   nowHook(): number; // Use for both Date.now() and new Date(),
-  // Supplied during Module instance creation instead of option
-  //   hasSourceTextAvailableHook(scriptOrModule): bool; // Used for censorship
-  resolveHook(name: string, referrer: FullSpecifier): FullSpecifier
-
-  // timing
-  // importHook delegates to importNowHook FIRST,
-  // they share a memo cache
-  // order to ensures importNow never allows async value of import
-  // to be accessed prior to any attached promise
-  importHook(fullSpec: FullSpecifier): Promise<StaticModuleRecord>;
-  importNowHook(fullSpec: FullSpecifier): StaticModuleRecord?;
-
-  // copy own props after return
-  importMetaHook(fullSpec: FullSpecifier): object
-
-  // e.g.: 'fr-FR' - Affects appropriate ECMA-402 APIs within Compartment
-  localeHook(): string;
-  // This is important to be able to override for deterministic testing and such
-  localTZAHook(): string;
-
-  // determines if the fn is acting as an "eval" function
-  isDirectEvalHook(evalFunctionRef: any): boolean;
-
-  // prep for trusted types non-string
-  canCompileHook(source: any, {
-    evaluator: functionRef, // can be a value from isDirectEvalHook
-    isDirect?: boolean
-  }): boolean; // need to allow mimicing CSP including nonces
+  // Every Compartment has a reference to a global environment record that in
+  // turn contains a new globalThis object, global contour, and three
+  // specialized intrinsic evaluators: eval, Function, and Compartment instances.
+  // The new globalThis object contains a subset of the JavaScript language
+  // intrinsics (to be defined in this proposal) and other globals must be
+  // "endowed" to the compartment explicitly with the `globals` option.
+  // All of these evaluators close over the compartment's global environment
+  // record such that they evaluate code in that global environment.
+  // When borrowGlobals is false, a the new Compartment gets a new global
+  // environment record.
+  // When borrowGlobals is true, the new Compartment will have the
+  // same global environment record as associated with the Compartment
+  // constructor used to construct the compartment.
+  // To borrow the globals of an arbitrary compartment, use that compartment's
+  // Compartment constructor, like
+  // new compartment.globalThis.Compartment({ borrowGlobals: true }).
+  borrowGlobals: boolean,
```

Comment:
How is it going to borrow globals meanwhile have the different evaluators?

Does it mean in this case, it is an exotic object?

Reply:
Yeah I think this paragraph could use a re-wording. I believe in the global re-use, there is no different evaluators, they are the same.

Reply:
Right, global environment records own evaluators, compartments refer to a global environment record and the only difference is the guest’s is the same as the host’s or if the guest gets a new one.

I’ll seek to improve upon this working in a subsequent PR. That probably will involve moving the description out of the sketch.